README.md

```diff
@@ -1,35 +1,65 @@
-primesieve
-==========
+primesieve-ruby
+===============
 [![Gem Version](https://badge.fury.io/rb/primesieve.svg)](http://badge.fury.io/rb/primesieve)
 
-primesieve is a gem wrapper for the [primesieve](http://primesieve.org/) prime number generator
+Ruby bindings (gem wrapper) for the [primesieve C++ library](https://github.com/kimwalisch/primesieve). Generates primes orders of magnitude faster than any pure Ruby code.
 
 Installing
-==========
+----------
+
 * Get primesieve
-  * On OSX: `brew install primesieve`
-  * [download here](http://primesieve.org/downloads/)
-  * [follow instructions to build from source](http://primesieve.org/build.html)
+  * __OS X:__ `brew install primesieve`
+  * [Build from source](https://github.com/kimwalisch/primesieve#build-instructions-unix-like-oses)
 
 * Get the primesieve gem
   * `gem install primesieve`
--or-
+__-or-__
   * add `gem 'primesieve'` to your Gemfile and run `bundle install`
 
 Usage
-=====
-First `require 'primesieve'`
-Then you have access to the `Primesieve` module which has all the functions of primesieve documented [here](http://primesieve.org/api/primesieve_8h.html) with the following exceptions:
-* functions requiring a type cannot be passed a type and will default to `UINT64_PRIMES`
-* functions requiring a `size_t *` cannot be passed that arg (an appropriatly sized Ruby array will be returned)
-* the function names leave off the `primesieve_` prefix
-  * for example, to call `primesieve_generate_primes`, use `Primesieve.generate_primes(start, stop)`
-* the functions `callback_primes` and `parallel_callback_primes` take a block rather than a callback function
-* `primesieve_free` is not implemented
-
-But how fast is it?
-===================
-Ruby 2.1.3 [Prime](http://www.ruby-doc.org/stdlib-2.0.0/libdoc/prime/rdoc/Prime.html) module vs. primesieve to generate the first 10,000,000 primes
+-----
+
+The syntax of the primesieve Ruby bindings is nearly identical to the
+syntax of the primesieve C++ library.
+
+```Ruby
+require 'primesieve'
+
+# Generate an array with the primes inside [0, 20]
+primes = Primesieve.generate_primes(0, 20)
+
+# Generate an array with the first 10 primes starting at 0
+primes = Primesieve.generate_n_primes(10, 0)
+
+# Get the 10th prime
+puts Primesieve.nth_prime(10, 0);
+
+# Count the primes below 10**9
+puts Primesieve.count_primes(0, 10**9)
+```
+
+Here is a [list of all available functions](ext/primesieve/extconf.rb).
+
+Multi-threading
+---------------
+
+Counting primes and prime k-tuplets and finding the nth prime can be done in parallel using multiple threads, just use the ```parallel_``` prefix and primesieve will use all your CPU cores.
```

Comment:
> The bindings are there but they don't actually use all CPU cores in parallel because of the Ruby GIL.

Are you sure about this? I tested it myself (on Saturday) on my quad-core system (Fedora 20/Linux) and all CPU cores were running at 100%!